docs/docs/concepts/subjects.mdx

```diff
@@ -7,70 +7,80 @@ RelationTuplePrism()
-can also refer to an object by using the empty relation. Effectively, this is
-interpreted as "any relation, even a non-existent one".
+follow some [best practices](../performance.mdx). As a special case, subject
+sets can also refer to an object by using the empty relation. Effectively, this
+is interpreted as "any relation, even a non-existent one".
 
 Subject sets also represent all intermediary nodes in
-[the graph of relations](./graph-of-relations).
+[the graph of relations](./graph-of-relations.mdx).
 
 ## Basic Example
 
 In the basic case an application uses the same subject identifiers as it uses
-internally, e.g. a username like `zepatrik` or UUIDv4 like
-`480158d4-0031-4412-9453-1bb0cdf76104`.
-
-Head over to the [basic full feature example](../examples/olymp-file-sharing) to
-see an example with some context.
-
-## Encoding Application Information within Keto Subjects
-
-Because the Keto client can use arbitrary strings as subjects, it is possible to
-encode additional information within the object. The only characters not allowed
-within the subject string are Ory Keto's delimiters `:#@`. Please refer to the
-documentation of your database for performance considerations, and the maximum
-allowed string length.
-
-For example, this can be used to implement a crude ABAC system by encoding
-attributes next to the actual subject ID. The application can then define
-relation tuples that reflect permissions depending on the value of attributes.
-It will have to encode the subject containing the attributes on each request.
-
-:::note
-
-The notation of key-value pairs within `<>` is completely arbitrary and just
-chosen here for readability.
-
-:::
+internally, e.g. a constant, unique username like `zepatrik` or preferably
+UUIDv4 like `480158d4-0031-4412-9453-1bb0cdf76104`.
+
+Head over to the
+[basic full feature example](../examples/olymp-file-sharing.mdx) to see an
+example with some context.
+
+## Advanced Example: Using Application Information within Keto Subjects
+
+Because the Keto client can use arbitrary strings as subjects, it is tempting to
+encode application data within the subject. **We strongly discourage this
+practice.** Instead, you should use a UUID to map application data to Keto
+subjects. This has a view advantages:
+
+| Advantage                                     | Keto's Limitation                                                         |
+| --------------------------------------------- | ------------------------------------------------------------------------- |
+| control the encoding of data                  | rejects the delimiters `:#@`                                              |
+| control the size of data                      | allows only up to 64 characters                                           |
+| single source of truth <br/> easy data update | stores relation tuples independently of each other <br/> no global update |
```

Comment:
Same as in the other document. UUID has advantages? But Keto has limitations? Don't really get it 😅

Reply:
I want to tell people why they should use UUIDs instead.

What do you think of

> This is required to ensure
>
> 1. single source of truth and easy data update
> 2. free choice of encoding
> 3. unlimited data size.

Reply:
Yeah, I think the current table is a bit confusing because it's not clear how the advantage applies to keto's limitation. I think a short list makes probably more sense. If you want, you can add a short description of why this is important in the same bullet point - e.g. `because subjects in Ory Keto can not contain :#@ in their names` or something like that